### ISSUE TITLE
How the loss function is called by registry?

### ISSUE THREAD

**Issue by sikaiyin** [facebookresearch/mmf#753]
## ❓ Questions and Help
Hi, I am conducting researches on custom loss function design. Basically, I want to map the sample_list['targets'] back through an embedding layer to calculate the loss during training(and this embedding layer parameters can be updated by pytorch).
I know that I can register a custom loss function in loss.py, but since I also need to change the network structure here with an additional embedding layer, I was wondering how to realize this part? Where does the loss calculation exactly happen in MMF? Thanks!

**Comment by ytsheng** [comment]
Hi @sikaiyin, thanks for using mmf!

I believe [this](https://github.com/facebookresearch/mmf/blob/master/mmf/models/base_model.py#L196) is what you are looking for. Yes, you do need to create a custom loss function in `loss.py`. Hope this helps!

**Comment by sikaiyin** [comment]
That helps! Thanks a lot!
